### PR TITLE
Add project phase status badges to the Data dictionary

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/20325

## Testing
**URL to test:** 

https://deploy-preview-1513--atd-moped-main.netlify.app/moped/dev/lookups#moped-phases

**Steps to test:**

There are now badges on the table instead of just the phase name. 

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- ~[ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

File: moped-editor/src/views/dev/LookupsView/RecordTable.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ import CircularProgress from "@mui/material/CircularProgress";
  * @returns { JSX } an MUI TableCell component with the row/column value
  */
 const WrappedTableCell = ({ row, column: { key, handler } }) => {
-  return <TableCell>{handler ? handler(row[key]) : row[key]}</TableCell>;
+  return <TableCell>{handler ? handler(row[key], row) : row[key]}</TableCell>;
```
Comment (chiaberry) [inline moped-editor/src/views/dev/LookupsView/RecordTable.js:17]: I added the row object into the handler call, but only one of our handler functions uses `row`, the one I just wrote. 

I can't decide if i should just leave the rest of the handler function definitions as is, or if i should update all of them to make explicit that they are getting an unneeded row object.

File: moped-editor/src/views/dev/LookupsView/settings.js [hunk 2]
```diff
@@ -40,6 +42,20 @@ const workTypeHandler = (workTypes) =>
     <div key={workType.moped_work_type.id}>{workType.moped_work_type.name}</div>
   ));
```
Comment (chiaberry) [inline moped-editor/src/views/dev/LookupsView/settings.js:43]: for example, should this be defined instead (expand to see line 37)
`const workTypeHandler = (workTypes, row) => `

File: moped-editor/src/views/dev/LookupsView/RecordTable.js (hunk 1 above)
Comment (mddilley) [reply moped-editor/src/views/dev/LookupsView/RecordTable.js:17]: hmmm, i'm going to read about this some. I think part of why this feels off is that some of the `handler` functions are returning JSX so they are function components in themselves. 🤔

**Review by frankhereford — APPROVED** [review]
I took a look at the deploy preview's data dictionary, and seeing the badges on the table drives home that this was a really good idea -- that is quite helpful.

Your code looks great to me, and I don't really have any idea about what the best route is re your question about the handler. I'll be real interested to see what you decide. 

All that said -- this is ready to be anchors-up, shipping-out.

File: moped-editor/src/views/dev/LookupsView/RecordTable.js (hunk 1 above)
Comment (johnclary) [reply moped-editor/src/views/dev/LookupsView/RecordTable.js:17]: this looks great to me! nbd to omit the `row` prop from the other handlers that do not access it. i think that is preferable to adding, for example, the `row` arg to the `workTypeHandler` function signature 🚢

Comment (mddilley) [reply moped-editor/src/views/dev/LookupsView/RecordTable.js:17]: yep, i was getting side-tracked with code beyond this diff. thanks for reining this back in, John! 🙏

**Review by mddilley — APPROVED** [review]
Great to see these badges! Thanks! 🚢 😎

**Review by tillyw — APPROVED** [review]
tested and the badges look great! i'm glad y'all came to a decision about `row` prop, i think that makes sense. thank you!

**Comment by johnclary** [comment]
This such a nice lil upgrade—great work!

**Pull request merged**